pkg/download/download.go

```diff
@@ -120,17 +120,80 @@ func downloadConfigFromEnvironment(fs afero.Fs, environment environment.Environm
 		util.Log.Info(" --- GETTING CONFIGS for %s", api.GetId())
 		jcreator := jsoncreator.NewJSONCreator()
 		ycreator := yamlcreator.NewYamlConfig()
-		errorAPI := createConfigsFromAPI(fs, api, token, path, client, jcreator, ycreator)
-		if errorAPI != nil {
-			util.Log.Error("error getting configs from API %v %v", api.GetId())
+
+		// Retrieves single object from legacy API
+		isLegacyApi := api.IsLegacyApi()
+		if isLegacyApi {
+			errorAPI := createConfigsFromLegacyAPI(fs, api, token, path, client, jcreator, ycreator)
+			if errorAPI != nil {
+				util.Log.Error("error getting configs from API %v %v", api.GetId())
+			}
+		} else {
+			errorAPI := createConfigsFromAPI(fs, api, token, path, client, jcreator, ycreator)
+			if errorAPI != nil {
+				util.Log.Error("error getting configs from API %v %v", api.GetId())
+			}
 		}
 	}
 	util.Log.Info("END downloading info %s", projectName)
 	return nil
 }
 
-func createConfigsFromAPI(fs afero.Fs, api api.Api, token string, fullpath string, client rest.DynatraceClient,
-	jcreator jsoncreator.JSONCreator, ycreator yamlcreator.YamlCreator) (err error) {
+func createConfigsFolder(
```

Comment:
as far as I can tell this is the same as done on line 207-208 for non legacy folder creation? 

assuming you pulled this into a method to avoid duplication, please also use the method there :)